lib/semian/net_http.rb

```diff
@@ -67,9 +76,11 @@ def reset_exceptions
     Semian::NetHTTP.reset_exceptions
 
     def raw_semian_options
-      options = semian_configuration
-      options = options.dup unless options.nil?
-      options
+      check_semian_configuration_changed
+      @raw_semian_options ||= begin
+        @raw_semian_options = Semian::NetHTTP.retrieve_semian_configuration(address, port)
+        @raw_semian_options = @raw_semian_options.dup unless @raw_semian_options.nil?
+      end
```

Comment:
The bug is that subsequent invocations of a function can change their return value. That's the expected behaviour. Anything else is magic.